tests/jobs/rft/test_trajectory.py

```diff
@@ -0,0 +1,103 @@
+import itertools
+import unittest
+import pytest
+import argparse
+
+from semeio.jobs.rft.trajectory import TrajectoryPoint, Trajectory
+from semeio.jobs.rft.zonemap import ZoneMap
+from semeio.jobs.rft.utility import parse_trajectory_line, load_trajectory_points
+
+
+class TrajectoryPointTest(unittest.TestCase):
+    def test_load_from_line(self):
+
+        for line, expected_zone in zip(["0 1 2.2 3", "0 1 2.2 3 zone"], [None, "zone"]):
+            point = TrajectoryPoint(*parse_trajectory_line(line))
+            self.assertEqual(point.utm_x, 0)
+            self.assertEqual(point.utm_y, 1)
+            self.assertEqual(point.measured_depth, 2.2)
+            self.assertEqual(point.true_vertical_depth, 3)
+            self.assertEqual(point.zone, expected_zone)
+
+    def test_invalid_load_from_line(self):
+        lines = ["1", "a b c", "a b c d e"]
+        errors = [
+            "Trajectory data file not on correct format",
+            "Trajectory data file not on correct format",
+            "Error: Failed to extract data from line",
+        ]
+
+        for line, error in zip(lines, errors):
+            with self.assertRaises(argparse.ArgumentTypeError) as err_context:
+                TrajectoryPoint(*parse_trajectory_line(line))
+            self.assertTrue(error in str(err_context.exception))
+
+    def test_validate_zone(self):
+        zones_at_k_value = {1: ["Zone1", "Zone2"], 3: ["Zone3"]}
+        zone_map = ZoneMap(zones_at_k_value)
+
+        k_values = [1, 3, 4]
+        expected_validation = [True, False, False]
+        for k_value, validation in zip(k_values, expected_validation):
+            point = TrajectoryPoint(0, 1, 2, 3, "Zone1")
+            point.set_ijk([0, 0, k_value])
+
+            self.assertFalse(point.valid_zone)
+            point.validate_zone(zone_map)
+
+            self.assertEqual(point.valid_zone, validation)
+
+        point = TrajectoryPoint(0, 1, 2, 3, "non existing")
+        point.set_ijk((0, 0, 1))
+        point.validate_zone(zone_map)
+        self.assertFalse(point.valid_zone)
+
+    def test_is_active(self):
+        ijks = [None, (0, 0, 0)]
+        pressures = [None, 1]
+        validations = [False, True]
+        for ijk, pressure, validation in itertools.product(
+            ijks, pressures, validations
+        ):
+            point = TrajectoryPoint(0, 1, 2, 3, "Zone1")
+            point.set_ijk(ijk)
+            point.pressure = pressure
+            point.valid_zone = validation
+            self.assertEqual(all([ijk, pressure, validation]), point.is_active())
+            self.assertEqual(
+                1 if all([ijk, pressure, validation]) else -1, point.get_pressure()
+            )
+
+
+class TestTrajectory(unittest.TestCase):
+    @pytest.fixture(autouse=True)
```

Comment:
If you're using unittest, wouldn't it be better to just use `def setUp`? Or just drop the `class` and use pure pytest?

Reply:
Yeh, this is a leftover. The previous repo used unittest, hence why it was written in unittest first. Started removing unittest from `test_gendata_rft`, an apparently that is how far I got :smile: 

Will have everything run pytest like the rest of this repo

Reply:
fixed